dask/array/routines.py

```diff
@@ -635,6 +635,26 @@ def digitize(a, bins, right=False):
     return a.map_blocks(np.digitize, dtype=dtype, bins=bins, right=right)
 
 
+# TODO: dask linspace doesn't support delayed values
+def _linspace_from_delayed(start, stop, num=50):
```

Comment:
How hard would this be to push into `linspace` in this PR? If it's not super easy, we can put it off, but if possible I'd like to avoid the need for this helper.

Reply:
I think it would be a moderate refactor of `linspace`. Most of the logic in `linspace` is around chunking based on `num`, so if `num` is delayed, chunking is impossible—the only option is 1 chunk, which could of course be large. I guess we could raise an error if `num` is delayed and `chunks != -1`?

If we did 1 chunk when _any_ of the parameters were delayed, it would be easy. But if we want to keep chunking when `start` and `stop` are delayed, this logic for calculating `blockstart` and `blockstop` would have to get delayed into the graph: https://github.com/dask/dask/blob/8d9c99d0870bf1c5ef05f5cce1ee0e0020872c56/dask/array/creation.py#L247-L255

which isn't hard, but is kind of a pain.

Reply:
mmk, fine with putting that off and leaving things as is then.